### PR TITLE
fix for close tab with multi path issue

### PR DESCRIPTION
**Pull request by Manikkumar1988**
#1259

### CODE REVIEW

File: lib/handlers/targetHandler.js [hunk 1]
```diff
@@ -40,18 +40,31 @@ const isMatchingRegex = function (target, targetRegex) {
...
+	}
+	return targetUrl;
+};
+
 const isMatchingUrl = function (target, targetUrl) {
   if (targetUrl instanceof RegExp) {
     return false;
   }
+
+  targetUrl = prependHttp(targetUrl);
```
Comment (sriv) [inline lib/handlers/targetHandler.js:56]: `prependHttp` is parsing the url again, while the url is getting parsed in line 58 in the calling method.

I wonder if line 61 can simply return `http` as the default host?

Comment (Manikkumar1988) [reply lib/handlers/targetHandler.js:56]:
```
const url = require('url');
console.log(url.parse('amazon.com'));
console.log(url.parse('http://amazon.com'));
```

> Url {
  protocol: null,
  slashes: null,
  auth: null,
  host: null,
  port: null,
  hostname: null,
  hash: null,
  search: null,
  query: null,
  pathname: 'amazon.com',
  path: 'amazon.com',
  href: 'amazon.com'
}
Url {
  protocol: 'http:',
  slashes: true,
  auth: null,
  host: 'amazon.com',
  port: null,
  hostname: 'amazon.com',
  hash: null,
  search: null,
  query: null,
  pathname: '/',
  path: '/',
  href: 'http://amazon.com/'
}

As we observe, parsing without protocol results in different output.

I see the point of operation being done twice but resultant is different with(or out) protocol, If performance is the main concern we can parse the targetUrl before passing it to isMatchingUrl().

Would that be way to go?

Comment (sriv) [reply lib/handlers/targetHandler.js:56]: wasn't thinking of performance, it's a simple operation when seen in context of the usecase.

I was thinking of the code duplication.

**Comment by sriv** [comment]
@Manikkumar1988 - I've approved https://github.com/getgauge/taiko/pull/1299, but I'd like this to go in the release as well.

File: lib/handlers/targetHandler.js (hunk 1 above)
Comment (Manikkumar1988) [reply lib/handlers/targetHandler.js:56]: If Code duplication is of concern, not sure how do we fix that!

It looks hacky to me, may be we can do normal string comparison if `targetUrl` has prefix of 'http'? It is another workaround to surpass `url.parse()`

Comment (sriv) [reply lib/handlers/targetHandler.js:56]: I get it. I think in that case we can live with this. String pattern match is worse.

Comment (sriv) [reply lib/handlers/targetHandler.js:56]: I get it. I think in that case we can live with this. String pattern match is worse.

**Pull request merged**